pkg/lib/bundle/validate.go

```diff
@@ -132,11 +170,159 @@ func (i imageValidator) ValidateBundle(directory string) error {
 
 	_, err = ValidateChannelDefault(annotations[ChannelsLabel], annotations[ChannelDefaultLabel])
 	if err != nil {
-		annotationErrors = append(annotationErrors, err)
+		validationErrors = append(validationErrors, err)
+	}
+
+	if len(validationErrors) > 0 {
+		return NewValidationError(validationErrors)
+	}
+
+	return nil
+}
+
+// ValidateBundleContent confirms that the CSV and CRD files inside the bundle
+// directory are valid and can be installed in a cluster. Other GVK types are
+// also validated to confirm if they are "kubectl-able" to a cluster meaning
+// if they can be applied to a cluster using `kubectl` provided users have all
+// necessary permissions and configurations.
+// Inputs:
+// manifestDir: the directory which all bundle manifests files are located
+// Outputs:
+// error: ValidattionError which contains a list of errors
+func (i imageValidator) ValidateBundleContent(manifestDir string) error {
+	var validationErrors []error
+
+	i.logger.Debug("Validating bundle contents")
+
+	mediaType, err := GetMediaType(manifestDir)
+	if err != nil {
+		validationErrors = append(validationErrors, err)
+	}
+
+	switch mediaType {
+	case HelmType, PlainType:
+		return nil
+	}
+
+	supportedTypes := map[string]string{
+		csvKind:                "",
+		crdKind:                "",
+		secretKind:             "",
+		clusterRoleKind:        "",
+		clusterRoleBindingKind: "",
+		serviceKind:            "",
+		serviceAccountKind:     "",
+		roleKind:               "",
+		roleBindingKind:        "",
 	}
 
-	if len(annotationErrors) > 0 || len(formatErrors) > 0 {
-		return NewValidationError(annotationErrors, formatErrors)
+	var csvName string
+	unstObjs := []*unstructured.Unstructured{}
+	csvValidator := v.ClusterServiceVersionValidator
+	crdValidator := v.CustomResourceDefinitionValidator
+
+	// Read all files in manifests directory
+	items, _ := ioutil.ReadDir(manifestDir)
+	for _, item := range items {
+		fileWithPath := filepath.Join(manifestDir, item.Name())
+		data, err := ioutil.ReadFile(fileWithPath)
+		if err != nil {
+			validationErrors = append(validationErrors, fmt.Errorf("Unable to read file %s in supported types", fileWithPath))
+			continue
+		}
+
+		dec := k8syaml.NewYAMLOrJSONDecoder(strings.NewReader(string(data)), 30)
+		k8sFile := &unstructured.Unstructured{}
+		if err := dec.Decode(k8sFile); err == nil {
```

Comment:
why don't we invert this to the typical `if err != nil` and have 
```
err := dec.Decode(k8sFile)
if err != nil {
  validationErrors = append(validationErrors, err)
  continue
}
```
since we are inside the for loop. that avoids some nesting